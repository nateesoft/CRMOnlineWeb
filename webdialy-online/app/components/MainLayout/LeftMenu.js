import React, { useState, useEffect } from 'react';
import { FormattedMessage } from 'react-intl';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import Divider from '@material-ui/core/Divider';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import LocalMallIcon from '@material-ui/icons/LocalMall';
import DnsRoundedIcon from '@material-ui/icons/DnsRounded';
import PeopleIcon from '@material-ui/icons/People';
import LockIcon from '@material-ui/icons/Lock';
import RecentActorsIcon from '@material-ui/icons/RecentActors';
import CardGiftcardIcon from '@material-ui/icons/CardGiftcard';
import Collapse from '@material-ui/core/Collapse';
import ExpandLess from '@material-ui/icons/ExpandLess';
import ExpandMore from '@material-ui/icons/ExpandMore';
import ButtonLink from 'components/ButtonLink';

const useStyles = makeStyles(() => ({
  itemActiveItem: {
    background: '#f1e6e2',
  },
}));

export default function LeftMenu(props) {
  const classes = useStyles();
  const { leftMenu = [], appConstants, scope, title } = props;
  const [open, setOpen] = useState(false);

  const leftMenuMaster = leftMenu.filter(
    item => item.menu_id === 'Master' || item.menu_id === 'Settings',
  );
  const leftMenuOther = leftMenu.filter(
    item => item.menu_id !== 'Master' && item.menu_id !== 'Settings',
  );

  const handleClick = () => {
    setOpen(!open);
  };

  useEffect(() => {
    if (
      [
        'Database',
        'Promotion',
        'Roles',
        'Branch',
        'Company',
        'Product',
        'Stock',
      ].includes(title)
    ) {
      setOpen(true);
    }
  }, []);

  return (
    <List style={{ background: '#fbf9f8' }}>
      {leftMenuOther.map(({ id, icon, to_path: to }) => (
        <ButtonLink to={appConstants.publicPath + to} key={`menu${id}`}>
          <ListItem
            key={id}
            button
            className={clsx(id === title && classes.itemActiveItem)}
          >
            <FormattedMessage id={`${scope}.menu${id}`}>
              {title => (
                <ListItemIcon title={title}>
                  {icon === 'DnsRoundedIcon' && <DnsRoundedIcon />}
                  {icon === 'CardGiftcardIcon' && <CardGiftcardIcon />}
                  {icon === 'LocalMallIcon' && <LocalMallIcon />}
                  {icon === 'PeopleIcon' && <PeopleIcon />}
                  {icon === 'LockIcon' && <LockIcon />}
                  {icon === 'RecentActorsIcon' && <RecentActorsIcon />}
                </ListItemIcon>
              )}
            </FormattedMessage>
            <ListItemText>
              <FormattedMessage id={`${scope}.menu${id}`} />
            </ListItemText>
          </ListItem>
        </ButtonLink>
      ))}
      <Divider />
      <ListItem button onClick={handleClick}>
        <ListItemIcon>
          <DnsRoundedIcon />
        </ListItemIcon>
        <ListItemText primary="ข้อมูลระบบ" />
        {open ? <ExpandLess /> : <ExpandMore />}
      </ListItem>
      <Collapse in={open} timeout="auto" unmountOnExit>
        <List component="div" style={{ marginLeft: '55px' }}>
          {leftMenuMaster.map(({ id, icon, to_path: to }) => (
            <ButtonLink to={appConstants.publicPath + to} key={`menu${id}`}>
              <ListItem
                key={id}
                button
                className={clsx(id === title && classes.itemActiveItem)}
              >
                <ListItemText>
                  <FormattedMessage id={`${scope}.menu${id}`} />
                </ListItemText>
              </ListItem>
            </ButtonLink>
          ))}
        </List>
      </Collapse>
      <Divider />
    </List>
  );
}

LeftMenu.propTypes = {
  leftMenu: PropTypes.array,
  appConstants: PropTypes.object,
  scope: PropTypes.string,
  title: PropTypes.string,
};
