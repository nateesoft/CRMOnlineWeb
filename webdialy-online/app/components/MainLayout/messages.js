/*
 * MainLayout Messages
 *
 * This contains all the text for the MainLayout component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.MainLayout';

export default defineMessages({
  headerLogin: {
    id: `${scope}.headerLogin`,
    defaultMessage: 'Log In',
  },
  headerLogout: {
    id: `${scope}.headerLogout`,
    defaultMessage: 'Logout',
  },
  menuTitleAccount: {
    id: `${scope}.menuTitleAccount`,
    defaultMessage: 'Account',
  },
  menuTitleOrders: {
    id: `${scope}.menuTitleOrders`,
    defaultMessage: 'Orders',
  },
  menuTitleRequestOrder: {
    id: `${scope}.menuTitleRequestOrder`,
    defaultMessage: 'Request Order',
  },
  menuTitleMembers: {
    id: `${scope}.menuTitleMembers`,
    defaultMessage: 'Members',
  },
  menuTitleSettings: {
    id: `${scope}.menuTitleSettings`,
    defaultMessage: 'Settings',
  },
  menuTitleMaster: {
    id: `${scope}.menuTitleMaster`,
    defaultMessage: 'Master',
  },
  menuOverview: {
    id: `${scope}.menuOverview`,
    defaultMessage: 'Overview',
  },
  menuProfile: {
    id: `${scope}.menuProfile`,
    defaultMessage: 'Profile',
  },
  menuShopping: {
    id: `${scope}.menuShopping`,
    defaultMessage: 'Shopping',
  },
  menuTrackOrder: {
    id: `${scope}.menuTrackOrder`,
    defaultMessage: 'Track Order',
  },
  menuCheckCartList: {
    id: `${scope}.menuCheckCartList`,
    defaultMessage: 'Check cart list',
  },
  menuMemberList: {
    id: `${scope}.menuMemberList`,
    defaultMessage: 'Member List',
  },
  menuUsePromotion: {
    id: `${scope}.menuUsePromotion`,
    defaultMessage: 'Use Promotion',
  },
  menuRoles: {
    id: `${scope}.menuRoles`,
    defaultMessage: 'Roles',
  },
  menuDatabase: {
    id: `${scope}.menuDatabase`,
    defaultMessage: 'Database',
  },
  menuCompany: {
    id: `${scope}.menuCompany`,
    defaultMessage: 'Company',
  },
  menuBranch: {
    id: `${scope}.menuBranch`,
    defaultMessage: 'Branch',
  },
  menuProduct: {
    id: `${scope}.menuProduct`,
    defaultMessage: 'Product',
  },
  menuStock: {
    id: `${scope}.menuStock`,
    defaultMessage: 'Stock',
  },
  menuPromotion: {
    id: `${scope}.menuPromotion`,
    defaultMessage: 'Promotion',
  },
});
