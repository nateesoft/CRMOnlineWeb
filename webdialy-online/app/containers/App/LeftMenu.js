import React from 'react';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import ListSubheader from '@material-ui/core/ListSubheader';
import DashboardIcon from '@material-ui/icons/Dashboard';
import ShoppingCartIcon from '@material-ui/icons/ShoppingCart';
import PeopleIcon from '@material-ui/icons/People';
import BarChartIcon from '@material-ui/icons/BarChart';
import LayersIcon from '@material-ui/icons/Layers';
import AssignmentIcon from '@material-ui/icons/Assignment';
import ExitToApp from '@material-ui/icons/ExitToApp';
import List from '@material-ui/core/List';
import { Link } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import Divider from '@material-ui/core/Divider';

const useStyles = makeStyles(theme => ({
  listMenu: {
    textDecoration: 'none',
    color: 'black',
  },
}));

const LeftMenu = props => {
  const classes = useStyles();

  const mainListItems = (
    <div>
      <Link to="/dashboard" className={classes.listMenu}>
        <ListItem button selected>
          <ListItemIcon>
            <DashboardIcon />
          </ListItemIcon>
          <ListItemText primary="Dashboard" />
        </ListItem>
      </Link>
      <ListItem button>
        <ListItemIcon>
          <ShoppingCartIcon />
        </ListItemIcon>
        <ListItemText primary="Orders" />
      </ListItem>
      <ListItem button>
        <ListItemIcon>
          <PeopleIcon />
        </ListItemIcon>
        <ListItemText primary="Customers" />
      </ListItem>
      <ListItem button>
        <ListItemIcon>
          <BarChartIcon />
        </ListItemIcon>
        <ListItemText primary="Reports" />
      </ListItem>
      <ListItem button>
        <ListItemIcon>
          <LayersIcon />
        </ListItemIcon>
        <ListItemText primary="Integrations" />
      </ListItem>
    </div>
  );

  const secondaryListItems = (
    <div>
      <ListSubheader inset>All Report</ListSubheader>
      <ListItem button>
        <ListItemIcon>
          <AssignmentIcon />
        </ListItemIcon>
        <ListItemText primary="Current month" />
      </ListItem>
      <ListItem button>
        <ListItemIcon>
          <AssignmentIcon />
        </ListItemIcon>
        <ListItemText primary="Last quarter" />
      </ListItem>
      <ListItem button>
        <ListItemIcon>
          <AssignmentIcon />
        </ListItemIcon>
        <ListItemText primary="Year-end sale" />
      </ListItem>
    </div>
  );

  const thirdListItems = (
    <div>
      <ListSubheader inset>Account Setting</ListSubheader>
      <Link to="/profile" className={classes.listMenu}>
        <ListItem button>
          <ListItemIcon>
            <AssignmentIcon />
          </ListItemIcon>
          <ListItemText primary="Profile" />
        </ListItem>
      </Link>
      <Link to="/logout" className={classes.listMenu}>
        <ListItem button>
          <ListItemIcon>
            <ExitToApp />
          </ListItemIcon>
          <ListItemText primary="Logout" />
        </ListItem>
      </Link>
    </div>
  );

  return (
    <div>
      <List>{mainListItems}</List>
      <Divider />
      <List>{secondaryListItems}</List>
      <Divider />
      <List>{thirdListItems}</List>
    </div>
  );
};

export default LeftMenu;
