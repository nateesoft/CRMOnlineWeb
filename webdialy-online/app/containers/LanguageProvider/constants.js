/*
 *
 * LanguageProvider constants
 *
 */
export const publicPath = process.env.REACT_APP_PUBLIC_PATH || '';
export const apiHostService = process.env.API_HOST_SERVICE || 'http://localhost:5000';

export const CHANGE_LOCALE = 'app/LanguageToggle/CHANGE_LOCALE';
