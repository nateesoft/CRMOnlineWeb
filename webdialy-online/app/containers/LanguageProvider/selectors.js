import { createSelector } from 'reselect';
import { initialState } from './reducer';

/**
 * Direct selector to the languageToggle state domain
 */
export const selectLanguage = state => state.language || initialState;

/**
 * Select the language locale
 */

export const makeSelectLocale = () =>
  createSelector(
    selectLanguage,
    languageState => languageState.locale,
  );
