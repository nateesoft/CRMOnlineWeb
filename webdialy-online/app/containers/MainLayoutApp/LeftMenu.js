import React, { useState } from 'react';
import { FormattedMessage } from 'react-intl';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import Divider from '@material-ui/core/Divider';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import LocalMallIcon from '@material-ui/icons/LocalMall';
import DnsRoundedIcon from '@material-ui/icons/DnsRounded';
import PeopleIcon from '@material-ui/icons/People';
import LockIcon from '@material-ui/icons/Lock';
import RecentActorsIcon from '@material-ui/icons/RecentActors';
import CardGiftcardIcon from '@material-ui/icons/CardGiftcard';
import Collapse from '@material-ui/core/Collapse';
import ExpandLess from '@material-ui/icons/ExpandLess';
import ExpandMore from '@material-ui/icons/ExpandMore';
import ButtonLink from 'components/ButtonLink';

const useStyles = makeStyles(() => ({
  container: {
    background: '#fbf9f8',
  },
  itemActiveItem: {
    background: '#f1e6e2',
  },
  listSubItem: {
    marginLeft: '55px',
  },
}));

export default function LeftMenu(props) {
  const classes = useStyles();
  const { leftMenu = [], appConstants, scope, location } = props;
  const [open, setOpen] = useState(false);

  const leftMenuMaster = leftMenu.filter(
    item => item.menu_id === 'Master' || item.menu_id === 'Settings',
  );
  const leftMenuOther = leftMenu.filter(
    item => item.menu_id !== 'Master' && item.menu_id !== 'Settings',
  );

  const handleClick = () => {
    setOpen(!open);
  };

  return (
    <List className={classes.container}>
      {leftMenuOther.map(({ id, icon, to_path: to }) => (
        <ButtonLink
          to={appConstants.publicPath + to.replace('shopping', 'shopping/new')}
          key={`menu${id}`}
        >
          <ListItem
            key={id}
            button
            className={clsx(location.pathname.includes(to) && classes.itemActiveItem)}
          >
            <FormattedMessage id={`${scope}.menu${id}`}>
              {tt => (
                <ListItemIcon title={tt}>
                  {icon === 'DnsRoundedIcon' && <DnsRoundedIcon />}
                  {icon === 'CardGiftcardIcon' && <CardGiftcardIcon />}
                  {icon === 'LocalMallIcon' && <LocalMallIcon />}
                  {icon === 'PeopleIcon' && <PeopleIcon />}
                  {icon === 'LockIcon' && <LockIcon />}
                  {icon === 'RecentActorsIcon' && <RecentActorsIcon />}
                </ListItemIcon>
              )}
            </FormattedMessage>
            <ListItemText>
              <FormattedMessage id={`${scope}.menu${id}`} />
            </ListItemText>
          </ListItem>
        </ButtonLink>
      ))}
      {leftMenuMaster.length > 0 && (
        <>
          <Divider />
          <ListItem button onClick={handleClick}>
            <ListItemIcon>
              <DnsRoundedIcon />
            </ListItemIcon>
            <ListItemText id="menuSystem" primary="ข้อมูลระบบ" />
            {open ? <ExpandLess /> : <ExpandMore />}
          </ListItem>
          <Collapse in={open} timeout="auto" unmountOnExit>
            <List component="div" className={classes.listSubItem}>
              {leftMenuMaster.map(({ id, to_path: to }) => (
                <ButtonLink to={`${appConstants.publicPath}${to}`} key={`menu${id}`}>
                  <ListItem
                    key={id}
                    button
                    className={clsx(location.pathname.includes(to) && classes.itemActiveItem)}
                  >
                    <ListItemText>
                      <FormattedMessage id={`${scope}.menu${id}`} />
                    </ListItemText>
                  </ListItem>
                </ButtonLink>
              ))}
            </List>
          </Collapse>
          <Divider />
        </>
      )}
    </List>
  );
}

LeftMenu.propTypes = {
  leftMenu: PropTypes.array,
  appConstants: PropTypes.object,
  scope: PropTypes.string,
  location: PropTypes.object,
};
