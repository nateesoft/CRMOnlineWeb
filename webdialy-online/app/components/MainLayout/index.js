import React, { memo, useState } from 'react';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import CssBaseline from '@material-ui/core/CssBaseline';
import Drawer from '@material-ui/core/Drawer';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import Container from '@material-ui/core/Container';
import MenuIcon from '@material-ui/icons/Menu';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import DoubleArrow from '@material-ui/icons/DoubleArrow';
import ExitToApp from '@material-ui/icons/ExitToApp';
import { Helmet } from 'react-helmet';
import { Button } from '@material-ui/core';
import SubMenu from 'components/SubMenu';
import * as appConstants from 'containers/App/constants';
import LocaleToggle from 'containers/LocaleToggle';
import ButtonLink from 'components/ButtonLink';
import { scope } from './messages';
import LeftMenu from './LeftMenu';

const drawerWidth = 240;

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
  },
  toolbar: {
    paddingRight: 24, // keep right padding when drawer closed
  },
  toolbarIcon: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: '0 8px',
    ...theme.mixins.toolbar,
    background: '#BE6C4E',
    color: 'white',
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    background: 'chocolate',
  },
  appBarShift: {
    marginLeft: drawerWidth,
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(['width', 'margin'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  menuButton: {
    marginRight: 36,
  },
  menuButtonHidden: {
    display: 'none',
  },
  title: {
    flexGrow: 1,
  },
  drawerPaper: {
    position: 'relative',
    whiteSpace: 'nowrap',
    width: drawerWidth,
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  drawerPaperClose: {
    overflowX: 'hidden',
    transition: theme.transitions.create('width', {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    width: theme.spacing(7),
    [theme.breakpoints.up('sm')]: {
      width: theme.spacing(9),
    },
  },
  appBarSpacer: theme.mixins.toolbar,
  content: {
    flexGrow: 1,
    height: '100vh',
  },
  container: {
    paddingTop: theme.spacing(1),
    paddingLeft: theme.spacing(1),
    paddingBottom: theme.spacing(4),
  },
  paper: {
    padding: theme.spacing(2),
    display: 'flex',
    overflow: 'auto',
    flexDirection: 'column',
  },
  fixedHeight: {
    height: 240,
  },
  divider: {
    marginTop: theme.spacing(2),
  },
  itemActiveItem: {
    background: '#f1e6e2',
  },
}));

const MainLayout = props => {
  const classes = useStyles();
  const { leftMenu } = props;
  const { profile } = props;
  const [open, setOpen] = useState(false);
  const handleDrawerOpen = () => {
    setOpen(true);
  };
  const handleDrawerClose = () => {
    setOpen(false);
  };

  const notMemberRole = profile && profile.member_role !== 'member';

  return (
    <div className={classes.root}>
      <Helmet>
        <title>{props.title}</title>
      </Helmet>
      <CssBaseline />
      <AppBar
        position="absolute"
        className={clsx(
          classes.appBar,
          open && notMemberRole && classes.appBarShift,
        )}
      >
        <Toolbar className={classes.toolbar}>
          {notMemberRole && (
            <IconButton
              edge="start"
              color="inherit"
              aria-label="open drawer"
              onClick={handleDrawerOpen}
              className={clsx(
                classes.menuButton,
                open && classes.menuButtonHidden,
              )}
            >
              <MenuIcon />
            </IconButton>
          )}
          <Typography
            component="span"
            color="inherit"
            noWrap
            className={classes.title}
          >
            {props.title}
          </Typography>
          <LocaleToggle />
          <ButtonLink to={`${appConstants.publicPath}/logout`}>
            <ExitToApp />
          </ButtonLink>
        </Toolbar>
      </AppBar>
      {notMemberRole && (
        <Drawer
          variant="permanent"
          classes={{
            paper: clsx(classes.drawerPaper, !open && classes.drawerPaperClose),
          }}
          open={open}
        >
          <div className={classes.toolbarIcon}>
            <Typography component="span" color="inherit">
              CRM Online
            </Typography>
            <IconButton onClick={handleDrawerClose}>
              <ChevronLeftIcon style={{ color: 'white' }} />
            </IconButton>
          </div>
          <Divider />
          <LeftMenu leftMenu={leftMenu} appConstants={appConstants} scope={scope} title={props.title} />
          <Button onClick={open ? handleDrawerClose : handleDrawerOpen}>
            {open ? (
              <span>
                <ChevronLeftIcon />
                ซ่อนเมนู
              </span>
            ) : (
              <DoubleArrow color="disabled" />
            )}
          </Button>
        </Drawer>
      )}
      <main className={classes.content}>
        <div className={classes.appBarSpacer} />
        <SubMenu {...props} />
        <Container maxWidth="lg" className={classes.container}>
          {props.children}
        </Container>
      </main>
    </div>
  );
};

MainLayout.propTypes = {
  leftMenu: PropTypes.array,
  profile: PropTypes.object,
  title: PropTypes.string,
  children: PropTypes.any,
};

export default memo(MainLayout);
