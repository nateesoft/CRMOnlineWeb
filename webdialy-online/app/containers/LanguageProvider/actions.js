/*
 *
 * LanguageProvider actions
 *
 */

import * as constants from './constants';

export function changeLocale(languageLocale) {
  return {
    type: constants.CHANGE_LOCALE,
    locale: languageLocale,
  };
}
