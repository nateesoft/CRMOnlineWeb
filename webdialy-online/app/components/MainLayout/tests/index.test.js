import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import { CssBaseline } from '@material-ui/core';

import LanguageProvider from 'containers/LanguageProvider';
import history from 'utils/history';
import configureStore from 'configureStore';
import { translationMessages } from 'i18n';

import MainLayout from '../index';
import LeftMenu from '../LeftMenu';

const initialState = {};
const store = configureStore(initialState, history);

describe('MainLayout components', () => {
  it('test main snapshot', () => {
    const component = renderer.create(
      <Provider store={store}>
        <LanguageProvider messages={translationMessages}>
          <ConnectedRouter history={history}>
            <CssBaseline />
            <MainLayout title="MainLayout">
              <div>Hello Main Layout</div>
            </MainLayout>
          </ConnectedRouter>
        </LanguageProvider>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('test LeftMenu snapshot', () => {
    const component = renderer.create(<LeftMenu />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
