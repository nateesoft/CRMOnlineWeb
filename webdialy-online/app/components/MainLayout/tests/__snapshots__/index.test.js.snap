// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout components test LeftMenu snapshot 1`] = `
<ul
  className="MuiList-root makeStyles-container-24 MuiList-padding"
/>
`;

exports[`MainLayout components test main snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionAbsolute MuiAppBar-colorPrimary makeStyles-appBar-4 MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-2 MuiToolbar-gutters"
    >
      <span
        className="MuiTypography-root makeStyles-title-8 MuiTypography-body1 MuiTypography-colorInherit MuiTypography-noWrap"
      >
        MainLayout
      </span>
      <div
        className="Wrapper-sc-1eh3f0x-0 hdXGaY"
      >
        <select
          className="makeStyles-textWhite-19 Select-qnzjgx-0 cQQWoM"
          onChange={[Function]}
          value="th"
        >
          <option
            value="en"
          >
            อังกฤษ
          </option>
          <option
            value="th"
          >
            ไทย
          </option>
        </select>
      </div>
      <a
        className="ButtonLink__StyledLink-sc-1pptk3-0 gOeyQX"
        href="/logout"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </a>
    </div>
  </header>
  <main
    className="makeStyles-content-12"
  >
    <div
      className="makeStyles-appBarSpacer-11"
    />
    <div
      className="MuiBottomNavigation-root makeStyles-layoutFixed-23"
    />
    <div
      className="MuiContainer-root makeStyles-container-13 MuiContainer-maxWidthLg"
    >
      <div>
        Hello Main Layout
      </div>
    </div>
  </main>
</div>
`;
